### ISSUE TITLE
Supporting password updates of mirrors

### ISSUE THREAD

**Issue by stephanpelikan** [dzikoysk/reposilite#1638]
### Request details

I've created a plugin which periodically fetches tokens for AWS codefactory using this command:
```
aws codeartifact get-authorization-token --domain my-domain --domain-owner  1234567 --query authorizationToken --output text
```

I'm not familar with Kotlin so I wrote it in Java. I can get the in-memory password property like this:

```
        final var auth = extensions()
                .getFacades()
                .stream()
                .filter(f -> f instanceof MavenFacade)
                .map(f -> (MavenFacade) f)
                .findFirst()
                .get()
                .getRepositories()
                .stream()
                .flatMap(r -> r.getMirrorHosts().stream())
                .map(m -> m.getConfiguration().getAuthorization())
                .filter(a -> a.getMethod() == AuthenticationMethod.BASIC)
                .filter(a -> a.getLogin().equals(user))
                .collect(Collectors.toList());
```

As a first try I updated the password-field using reflection since there is no public setter, but it seems not to work or to be used because I see 401 on using the mirrored repo which means the dummy password configured is still in use and not the valid one set via reflection.

However, is it necessary to update the shared config? If yes, how can this be done? If no, do you have an idea what's wrong?

Thanks ❤️ 
Stephan

**Comment by dzikoysk** [comment]
Interesting! Plugin API is quite new, so it may lack of some features, but let's see what we can do with it. As far as I see, password is used directly from configuration object, so in theory it should work if you've update the `MirrorCredentials.password` field used 🤔 

Speaking of the "correct" way to do this, you should update Maven plugin settings:

```kotlin
val sharedConfigurationFacade = facade<SharedConfigurationFacade>()
val currentMavenConfigurationReference = sharedConfigurationFacade.getDomainSettings<MavenSettings>()
val currentMavenConfiguration = currentMavenConfigurationReference.get()

sharedConfigurationFacade.updateSharedSettings("maven", currentMavenConfiguration.copy(
    repositories = currentMavenConfiguration.repositories.map { repository ->
        when (repository.id) {
            "releases" -> 
                repository.copy(
                    proxied = repository.proxied.map { mirror ->
                        mirror.copy(
                            authorization = mirror.authorization?.copy(
                                password = "New password!"
                            )
                        )
                    }
                )
            else -> repository
        }
    }
))
```

I really recommend to use Kotlin as it's quite complex immutable data structure to update, so it'll be slightly painful to write in Java 😅

**Comment by stephanpelikan** [comment]
Thank you for your answer. It seems my plugin works. I have to test it with several Maven-projects and for some reasons the repo-id is sometimes different but very similar and I didn't recognized that. So I got 401 because the credentials were not found due to that different repo-id. I misinterpreted this response in the way that I thought authentication fails after a while, but it was always fine. Sorry for that confusion and thank you for helping me ❤️ 😄.

By-the-way: Great tool 👍
